lib/r10k/source/base.rb

```diff
@@ -23,16 +23,27 @@ class R10K::Source::Base
   # @param options [Hash] An additional set of options for this source. The
   #   semantics of this hash may depend on the source implementation.
   #
-  # @option options [Boolean] :prefix Whether to prefix the source name to the
+  # @option options [Boolean | String] :prefix Whether to prefix the source name to the
   #   environment directory names. All sources should respect this option.
   #   Defaults to false.
   def initialize(name, basedir, options = {})
     @name    = name
     @basedir = basedir
-    @prefix  = options.delete(:prefix)
+    @prefix  = derive_prefix(name,options.delete(:prefix))
     @options = options
   end
 
+  def derive_prefix(name,prefix)
+    if prefix == true || prefix == 'true'
```

Comment:
Corrected documentation.

I see what you mean about truthy values.   The fix here would be to check 

`if (prefix.is_a?(TrueClass) && prefix)` 

 What's the best practice here?   I don't mind changing it, let me know what is best.